kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/RecordEncryptionFilter.java

```diff
@@ -222,7 +208,26 @@ private CompletionStage<List<PartitionData>> maybeDecodePartitions(String topicN
             if (!(partitionData.records() instanceof MemoryRecords)) {
                 throw new IllegalStateException();
             }
-            result.add(maybeDecodeRecords(topicName, partitionData, (MemoryRecords) partitionData.records(), context));
+            var stage = maybeDecodeRecords(topicName, partitionData, (MemoryRecords) partitionData.records(), context)
+                    .exceptionallyCompose(t -> {
+                        var cause = t.getCause();
+                        if (cause instanceof UnknownKeyException) {
+                            log.atWarn()
+                                    .setMessage("Failed to decrypt record in topic-partition {}-{} owing to key not found condition. "
```

Comment:
unfortunately, we don't have sufficient information to log the offset where the problem occured.